File: 1/1.6/Python/1.6.py
def compressString(s: str) -> str:
    compressed = ""
    counter = 1
    prev_c = ''
    for c in s:
        if c == prev_c:
            counter = counter + 1
        else:
            compressed = compressed + prev_c + str(counter)
            counter = 1
            prev_c = c
    compressed = compressed + s[-1] + str(counter)
    compressed = compressed[1:]

    if len(compressed) >= len(s):
        return s
    else:
        return compressed


if __name__ == '__main__':
    print(compressString("abc"))
    print(compressString("aabc"))
    print(compressString("Aabc"))
    print(compressString("aaaabbbcc"))
    print(compressString("aaAabbBcc"))
    print(compressString("aaAabbbBcc"))
    print(compressString("aaAabbbbBcc"))
    print(compressString("aaAabbbbbBcc"))
    print(compressString("aaAabbbbbbBcc"))

    

File: 1/1.6/Go/1.6.go
package main

import (
	"fmt"
	"strconv"
)

func compressString(s string) string {
	compressed := ""
	counter := 1
	prev_c := ""
	for _, c := range []rune(s) {
		if string(c) == prev_c {
			counter += 1
		} else {
			compressed = compressed + prev_c + strconv.Itoa(counter)
			counter = 1
			prev_c = string(c)
		}
	}
	compressed += string(s[len(s)-1]) + strconv.Itoa(counter)
	compressed = compressed[1:]

	if len(compressed) >= len(s) {
		return s
	} else {
		return compressed
	}
}

func main() {
	fmt.Println(compressString("abc"))
	fmt.Println(compressString("aabc"))
	fmt.Println(compressString("Aabc"))
	fmt.Println(compressString("aaaabbbcc"))
	fmt.Println(compressString("aaAabbBcc"))
	fmt.Println(compressString("aaAabbbBcc"))
	fmt.Println(compressString("aaAabbbbBcc"))
	fmt.Println(compressString("aaAabbbbbBcc"))
	fmt.Println(compressString("aaAabbbbbbBcc"))
}
